### PR TITLE
Fix incorrect processing of Iterable extended classes.

### LINKED ISSUE #753: Page should not be documented as an Iterable (Array)
### Expected Behavior

When a controller returns a `Page<Something>` (or `Mono<Page<Something>>`) the generated schema is an array of Something, because `io.micronaut.data.model.Page` is an `Iterable`.
But Page fields are then omited/missing.

### Actual Behaviour

Page schema should contains Page fields like totalSize, totalPages, size, offset, ...

### Steps To Reproduce

See this demo project : https://github.com/ggerbaud/demo-micronaut/tree/demo/openapi/page
mvn compile
then see target/classes/META-INF/swagger/*.yml file

### Environment Information

_No response_

### Example Application

https://github.com/ggerbaud/demo-micronaut/tree/demo/openapi/page

### Version

3.5.2 and previous

### PR DESCRIPTION
**Pull request by altro3**
Fixed #753

### CODE REVIEW

**Comment by graemerocher** [comment]
Needs conflicts resolving. Thanks.

**Comment by altro3** [comment]
@graemerocher All conflicts resolved

File: openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java [hunk 1]
```diff
@@ -1755,7 +1761,11 @@ private void populateSchemaProperties(OpenAPI openAPI, VisitorContext context, E
 
     private void processPropertyElements(OpenAPI openAPI, VisitorContext context, Element type, Schema schema, List<? extends TypedElement> publicFields, List<MediaType> mediaTypes) {
         for (TypedElement publicField : publicFields) {
-            if (publicField.isAnnotationPresent(JsonIgnore.class) || publicField.isAnnotationPresent(Hidden.class)) {
+            AnnotationValue<io.swagger.v3.oas.annotations.media.Schema> fieldSchemaAnn = publicField.getAnnotation(io.swagger.v3.oas.annotations.media.Schema.class);
+            boolean isHidden = fieldSchemaAnn != null && fieldSchemaAnn.get("hidden", Boolean.class).orElse(false);
```
Comment (graemerocher) [inline openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java:1765]: can be simplified to:

```java
boolean isHidden = publicField.getAnnotationMetdata().booleanValue(Schema.class, "hidden").orElse(false)
```

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java (hunk 1 above)
Comment (altro3) [reply openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java]: @graemerocher fixed

**Comment by graemerocher** [comment]
Thanks for the contribution!

**Pull request merged**